ui/server/src/main/scala/pl/touk/nussknacker/ui/api/PublicResources.scala

```diff
@@ -0,0 +1,34 @@
+package pl.touk.nussknacker.ui.api
+
+import akka.http.scaladsl.server.directives.SecurityDirectives
+import akka.http.scaladsl.server.{Directives, Route}
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.ProcessingTypeData.ProcessingType
+
+import scala.concurrent.ExecutionContext
+
+class PublicResources(config: Config, modelData: Map[ProcessingType, ModelData])(implicit ec: ExecutionContext)
+  extends Directives with FailFastCirceSupport with LazyLogging with RouteWithoutUser with SecurityDirectives {
+
+  import io.circe.syntax._
+  import net.ceedubs.ficus.Ficus._
+
+  def route(): Route = {
+    pathPrefix("app") {
```

Comment:
Now we have two resources prefixed by "app"? How it supposed to work? IMO we should create one AppResources and create it from public and non-public resource paths... or create new resource prefix `/public`. The first option I think is nicer...

Reply:
There is an joining paths pattern. Hmm I can extends from RouteWithUser and RouteWithoutUser, and implements two kind of method rout- one with user second without.

Reply:
I'm not convinced that we want this joining pattern to use... Now you have two paths prefix that someone might change separately introducing regression bug. Also path hierarchy is not visible. IMO would be better to have one route() but inside of this have two private route parts...

Reply:
I'v added some comments.